### PR TITLE
Monitoring docs changes between 4.0 and 4.1 + Add Telemetry docs

### PR DESCRIPTION
**Pull request by rh-max**

### CODE REVIEW

**Comment by vikram-redhat** [comment]
@rh-max - can you change the title of this PR to better reflect what it is about?

Also, looks like a minor issue with the build.

**Comment by rh-max** [comment]
Note: the Telemetry docs, while not part of the Monitoring docs, are included in this PR as well. Here's the Telemetry Jira: https://jira.coreos.com/browse/OSDOCS-401

File: monitoring/cluster-monitoring/managing-alerts-using-monitoring-ui.adoc [hunk 1]
```diff
@@ -1,16 +1,20 @@
...
+include::modules/monitoring-contents-of-monitoring-ui.adoc[leveloffset=+1]
 include::modules/monitoring-getting-information-about-alerts-and-alerting-rules.adoc[leveloffset=+1]
 include::modules/monitoring-silencing-alerts.adoc[leveloffset=+1]
 include::modules/monitoring-getting-information-about-silences.adoc[leveloffset=+1]
 include::modules/monitoring-editing-silences.adoc[leveloffset=+1]
 include::modules/monitoring-expiring-silences.adoc[leveloffset=+1]
 
+.Next steps
+
+xref:../../monitoring/cluster-monitoring/prometheus-alertmanager-and-grafana.adoc#prometheus-alertmanager-and-grafana[Access the Prometheus, Alertmanager, and Grafana.]
```
Comment (vikram-redhat) [inline monitoring/cluster-monitoring/managing-alerts-using-monitoring-ui.adoc]: @rh-max If you change the anchor text to: #accessing-prometheus from: #prometheus-alertmanager-and-grafana the build will pass for this particular error.

There is a big long explanation about this, and am happy to go into it when we see each other :)

Comment (rh-max) [reply monitoring/cluster-monitoring/managing-alerts-using-monitoring-ui.adoc]: @vikram-redhat No comment. :)

File: modules/telemetry-disabling-telemetry.adoc [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  metadata:
+    name: cluster-monitoring-config
+    namespace: openshift-monitoring
+  data:
+    config.yaml: |+
+      *telemeterClient:*
+        *enabled: false*
+----
+
+.Additional resources
```
Comment (kalexand-rh) [inline modules/telemetry-disabling-telemetry.adoc]: Remove this section.

File: modules/telemetry-showing-data-collected-from-the-cluster.adoc [hunk 3]
```diff
@@ -0,0 +1,30 @@
+// Module included in the following assemblies:
+//
+// * telemetry/showing-data-collected-by-telemetry.adoc
+
+[id='showing-data-collected-from-the-cluster-{context}']
+= Showing data collected from the cluster
+
+You can see the time series captured by Telemetry.
```
Comment (kalexand-rh) [inline modules/telemetry-showing-data-collected-from-the-cluster.adoc]: What time series?

File: modules/telemetry-showing-data-collected-from-the-cluster.adoc [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+.Prerequisites
+
+* Log in as a user with administrative privileges for the cluster.
+
+.Procedure
+
+. Find the URL for the Prometheus service running in the OpenShift cluster:
```
Comment (kalexand-rh) [inline modules/telemetry-showing-data-collected-from-the-cluster.adoc]: s/running/that runs
s/OpenShift/{product-title}

File: modules/telemetry-showing-data-collected-from-the-cluster.adoc [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+// * telemetry/showing-data-collected-by-telemetry.adoc
+
+[id='showing-data-collected-from-the-cluster-{context}']
+= Showing data collected from the cluster
+
+You can see the time series captured by Telemetry.
+
+.Prerequisites
+
+* Log in as a user with administrative privileges for the cluster.
```
Comment (kalexand-rh) [inline modules/telemetry-showing-data-collected-from-the-cluster.adoc]:
```
* Install the OpenShift Command-line Interface (CLI), commonly known as `oc`.
* You must log in to the cluster with a user that has the `cluster-admin` role.
```

File: modules/telemetry-showing-data-collected-from-the-cluster.adoc [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+.Procedure
+
+. Find the URL for the Prometheus service running in the OpenShift cluster:
++
```
Comment (kalexand-rh) [inline modules/telemetry-showing-data-collected-from-the-cluster.adoc]: All the codeblocks need to be enclosed with:
```
----
----
```

File: modules/telemetry-showing-data-collected-from-the-cluster.adoc [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+  $ oc get route prometheus-k8s -n openshift-monitoring -o jsonpath="{.spec.host}"
+
+. Navigate to the URL.
+
+. Run this query:
```
Comment (kalexand-rh) [inline modules/telemetry-showing-data-collected-from-the-cluster.adoc]: Where do I run this query? Somewhere on the page at the URL?

File: modules/telemetry-showing-data-collected-from-the-cluster.adoc [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
++
+This query replicates the request that Telemetry makes against a running {product-title} cluster's Prometheus service and returns the full set of time series captured by Telemetry.
+
+.Additional resources
+
+* See the link:https://github.com/openshift/telemeter/blob/master/docs/data-collection.md[Telemetry repository] for the list of metrics used in the query.
```
Comment (kalexand-rh) [inline modules/telemetry-showing-data-collected-from-the-cluster.adoc]: Support has to approve all GitHub links. Even internal ones.

File: modules/telemetry-what-information-is-collected.adoc [hunk 9]
```diff
@@ -0,0 +1,29 @@
+// Module included in the following assemblies:
+//
+// * telemetry/telemetry.adoc
+
+[id='what-information-is-collected-{context}']
+= What information is collected
+
+Telemetry does not collect, and will never collect identifying information like user names, passwords, or the names or addresses of user resources.
```
Comment (kalexand-rh) [inline modules/telemetry-what-information-is-collected.adoc]: s/collect/collect,

File: modules/telemetry-what-information-is-collected.adoc [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+
+Primary information collected includes:
+
+* Number of updates available per cluster
```
Comment (kalexand-rh) [inline modules/telemetry-what-information-is-collected.adoc]: s/nodes/machines everywhere in this list

File: modules/telemetry-what-information-is-collected.adoc [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+* Channel and image repository used for an update
+* Number of errors that occurred during an update
+* Progress information of an update that is running
+* Number of nodes per cluster
+* Number of CPU cores and size of RAM of the nodes
+* Size of etcd
```
Comment (kalexand-rh) [inline modules/telemetry-what-information-is-collected.adoc]: Please clarify.

File: modules/telemetry-what-information-is-collected.adoc [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+* Number of CPU cores and RAM used per node type (infra or master)
```
Comment (kalexand-rh) [inline modules/telemetry-what-information-is-collected.adoc]: There are no infra nodes by default. Don't use parentheses.

File: modules/telemetry-what-information-is-collected.adoc [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+* Number of CPU cores and RAM used per cluster
+* Utilization of {product-title} framework components per cluster
```
Comment (kalexand-rh) [inline modules/telemetry-what-information-is-collected.adoc]: s/Utilization/use

File: modules/telemetry-what-information-is-collected.adoc [hunk 14]
```diff
@@ -0,0 +1,29 @@
...
+* Version of the {product-title} installation
```
Comment (kalexand-rh) [inline modules/telemetry-what-information-is-collected.adoc]: s/installation/cluster ?

File: modules/telemetry-what-information-is-collected.adoc [hunk 15]
```diff
@@ -0,0 +1,29 @@
...
+* Health, condition, and status for any {product-title} framework component installed on the cluster, for example Cluster Version Operator, Cluster Monitoring, Image Registry, Elasticsearch for Logging
```
Comment (kalexand-rh) [inline modules/telemetry-what-information-is-collected.adoc]: s/installed/that is installed

Comment (kalexand-rh) [reply modules/telemetry-what-information-is-collected.adoc]: s/Elasticsearch/ and Elasticsearch

File: modules/telemetry-what-information-is-collected.adoc [hunk 16]
```diff
@@ -0,0 +1,29 @@
...
+* A unique random identifier generated during installation
```
Comment (kalexand-rh) [inline modules/telemetry-what-information-is-collected.adoc]: s/generated/that is generated

File: modules/telemetry-what-information-is-collected.adoc [hunk 17]
```diff
@@ -0,0 +1,29 @@
...
+* Name of the platform {product-title} is deployed on, such as AWS
```
Comment (kalexand-rh) [inline modules/telemetry-what-information-is-collected.adoc]: s/AWS/Amazon Web Services

File: monitoring/cluster-monitoring/configuring-the-monitoring-stack.adoc [hunk 18]
```diff
@@ -27,13 +27,19 @@ include::modules/monitoring-moving-monitoring-components-to-different-nodes.adoc
 
 Running cluster monitoring with persistent storage means that your metrics are stored to a Persistent Volume and can survive a pod being restarted or recreated. This is ideal if you require your metrics or alerting data to be guarded from data loss. For production environments, it is highly recommended to configure persistent storage.
 
+[IMPORTANT]
+====
+In {product-title} 4.1 deployed on bare metal, Prometheus and Alertmanager cannot have persistent storage and are ephemeral only. For use cases other than bare metal, you must use dynamic provisioning to use persistent storage.
```
Comment (kalexand-rh) [inline monitoring/cluster-monitoring/configuring-the-monitoring-stack.adoc]: s/4.1/{product-version}

Is this just bare metal, or is it vSphere, too?

Why can't you have persistent storage for it?

File: telemetry/about-telemetry.adoc [hunk 19]
```diff
@@ -0,0 +1,17 @@
+[id="about-telemetry"]
+= About Telemetry
+include::modules/common-attributes.adoc[]
+:context: telemetry
+
+toc::[]
+
+Telemetry collects anonymized aggregated information about:
+
+* the size of an OpenShift cluster
```
Comment (kalexand-rh) [inline telemetry/about-telemetry.adoc]: Initial caps for all lis

s/OpenShift/{product-title}
^ always

File: telemetry/about-telemetry.adoc [hunk 20]
```diff
@@ -0,0 +1,17 @@
...
+* the health and status of OpenShift components
+* utilization of OpenShift components
```
Comment (kalexand-rh) [inline telemetry/about-telemetry.adoc]: s/utilization/use

What's the difference between features and components?

What's the difference between this list and the list in the included module?

File: telemetry/opting-out-of-telemetry.adoc [hunk 21]
```diff
@@ -0,0 +1,21 @@
+[id="opting-out-of-telemetry"]
+= Opting out of Telemetry
+include::modules/common-attributes.adoc[]
+:context: telemetry
+
+toc::[]
+
+You might need to opt out of Telemetry for your cluster.
+
+[NOTE]
```
Comment (kalexand-rh) [inline telemetry/opting-out-of-telemetry.adoc]: This note needs to be conditionalized to only display in OSD. Double-check with Vikram.

File: telemetry/opting-out-of-telemetry.adoc [hunk 22]
```diff
@@ -0,0 +1,21 @@
...
+====
+On OpenShift Dedicated, Telemetry is always enabled. You cannot opt out.
+====
+
+.Prerequisites
+
+* An installed `oc` CLI tool.
```
Comment (kalexand-rh) [inline telemetry/opting-out-of-telemetry.adoc]: We're changing the way that we use this in prereqs. Jason has an open PR with the details.

File: telemetry/showing-data-collected-by-telemetry.adoc [hunk 23]
```diff
@@ -0,0 +1,10 @@
+[id="showing-data-collected-by-telemetry"]
+= Showing data collected by Telemetry
+include::modules/common-attributes.adoc[]
+:context: telemetry
+
+toc::[]
+
+An administrator might need to understand what metrics are collected by Telemetry.
```
Comment (kalexand-rh) [inline telemetry/showing-data-collected-by-telemetry.adoc]: Maybe
> You can review the metrics that Telemetry collects

**Review by kalexand-rh — COMMENTED** [review]
I have some suggestions.

**Comment by kalexand-rh** [comment]
@vikram-redhat, this change adds a new bucket and some GitHub links. PTAL?

@rh-max, make sure that you squash your commits.

File: modules/telemetry-disabling-telemetry.adoc [hunk 24]
```diff
@@ -0,0 +1,32 @@
+// Module included in the following assemblies:
+//
+// * telemetry/opting-out-of-telemetry.adoc
+
+[id="disabling-telemetry-{context}"]
+= Disabling Telemetry
+
+You can disable Telemetry.
```
Comment (kalexand-rh) [inline modules/telemetry-disabling-telemetry.adoc]: We need a warning after the intro. @aheslin 
```
[IMPORTANT]
====
If you disable Telemetry, your cluster becomes unsupported and might not be able to update.
====

Comment (kalexand-rh) [reply modules/telemetry-disabling-telemetry.adoc]: @rh-max, please add this note.

**Comment by rh-max** [comment]
Thanks for the review Kathryn! I've implemented most of the feedback, plus awaiting input from the team regarding some of the items.

I have two outstanding things:

- Per feedback from the team, I did not add the note on Telemetry you requested. Instead, we are adding this:
https://github.com/openshift/openshift-docs/blob/24c63b1f2cfc467ec4cf271800ab01500189d6c2/modules/telemetry-consequences-of-disabling-telemetry.adoc
If your request and wording came from a solid source and can't be discarded, let me know please.

- Regarding the overlap between sections on moving components inside monitoring/ and machine_management/.
https://docs.openshift.com/container-platform/4.1/monitoring/cluster-monitoring/configuring-the-monitoring-stack.html#moving-monitoring-components-to-different-nodes_configuring-monitoring
https://docs.openshift.com/container-platform/4.1/machine_management/creating-infrastructure-machinesets.html#infrastructure-moving-monitoring_creating-infrastructure-machinesets
Regarding the one in monitoring/, I think it makes sense to keep it, so that the monitoring section is kept comprehensive. I don't know about the one inside machine_management/. If it's kept, then we should probably leave only one module to be included in both places. What do you think?

**Comment by kalexand-rh** [comment]
@rh-max, thanks!

re: notes - that's fine. We do need to link to the telemetry section from the install docs so that users can see how to disable it if they must, per @reestr.

re: overlapping modules - both assemblies do need the content, and we need to figure out how to make the module work in both locations. It doesn't have to happen on this PR or before GA, but it needs to happen at some point. If you don't want to do it on this PR, please open a JIRA on our board and add it to the CIT epic so we can track the requirement.

**Comment by vikram-redhat** [comment]
@rh-max - can you make sure that this PR is merged no later than Tuesday, COB, your time? IF there are other things to come for it we will just have to address them as they become available, but I need this PR to be merged so that the Telemetry bucket is available for the customer portal.

I believe QE has been done on this; make sure to squash the commits.

@kalexand-rh - thanks for reminder about the GitHub links; On a separate thread @sferich888 has rejected these so @rh-max will need to take them out.

**Pull request merged**